### PR TITLE
Include virtual agent to build pack & fix remote APIM connecting issue

### PR DESCRIPTION
**Pull request by charithag**

### CODE REVIEW

File: modules/distribution/pom.xml [hunk 1]
```diff
@@ -325,6 +325,11 @@
 											dir="target/wso2carbon-core-${carbon.kernel.version}/samples/raspberrypi"
 											includes="build.xml"/>
 								</subant>
+								<subant target="deploy">
+									<fileset
+											dir="target/wso2carbon-core-${carbon.kernel.version}/samples/virtual_firealarm"
+											includes="build.xml"/>
```
Comment (DilanUA) [inline modules/distribution/pom.xml:331]: fix indentation issues.

File: modules/samples/virtual_firealarm/build.xml [hunk 2]
```diff
@@ -23,7 +23,8 @@
                 <![CDATA[
     <IotDeviceTypeConfig type="${sample_type}">
 		<DatasourceName>jdbc/${data_source}</DatasourceName>
-		<ApiApplicationName>virtual_firealarm</ApiApplicationName>
+		<!--TODO: Uncomment this to work with apim @ 192.168.67.21 -->
```
Comment (DilanUA) [inline modules/samples/virtual_firealarm/build.xml:26]: remove this TODO as it seems irrelevant.

File: modules/distribution/pom.xml (hunk 1 above)
Comment (charithag) [inline modules/distribution/pom.xml:331]: Can't fix as this code is obsolete and no longer exists.

File: modules/samples/virtual_firealarm/build.xml (hunk 2 above)
Comment (charithag) [inline modules/samples/virtual_firealarm/build.xml:26]: Can't fix as this code is obsolete and no longer exists.
